src/hotspot/share/utilities/debug.cpp

```diff
@@ -480,6 +483,41 @@ extern "C" JNIEXPORT void pp(void* p) {
     oop obj = cast_to_oop(p);
     obj->print();
   } else {
+#if INCLUDE_NMT
+    // With NMT
+    if (MemTracker::enabled()) {
+      const NMT_TrackingLevel tracking_level = MemTracker::tracking_level();
+      // Check if it is in a known mmap'd memory region
+      const ReservedMemoryRegion* rgn = VirtualMemoryTracker::find_region(p);
+      if (rgn != nullptr) {
+        tty->print_cr(PTR_FORMAT " in mmap'd memory region [" PTR_FORMAT " - " PTR_FORMAT "] by %s",
+          p2i(p), p2i(rgn->base()), p2i(rgn->base() + rgn->size()), rgn->flag_name());
+        if (tracking_level == NMT_detail) {
+          rgn->call_stack()->print_on(tty);
```

Comment:
Idea for later cleanup. In detail mode, malloc headers have no call stack (NULL) whereas regions have an empty stack (NativeCallStack::_empty_stack) ? If yes, may be nice to unify this behavior. 

Maybe also remove checks like this here, for MemTracker::tracking_level=detail, in favor of checking if rgn->call_stack() is NULL. That would feel more consistent and allow us (if we ever wanted) to have regions with and without stack.

Reply:
Yes. I noticed the inconsistent of call stack handling and yes, I would like to defer to other CR.